File: docs/components_page/components/toast/auto_dismiss.jl
using DashBootstrapComponents, DashHtmlComponents

toast = html_div([
    dbc_button(
        "Open toast",
        id = "auto-toast-toggle",
        color = "primary",
        class_name = "mb-3",
        n_clicks = 0,
    ),
    dbc_toast(
        [html_p("This is the content of the toast", class_name = "mb-0")],
        id = "auto-toast",
        header = "This is the header",
        icon = "primary",
        duration = 4000,
    ),
]);


callback!(app, Output("auto-toast", "is_open"), Input("auto-toast-toggle", "n_clicks")) do n
    return true
end;

File: docs/components_page/components/alert/auto_dismiss.py
import dash_bootstrap_components as dbc
from dash import Input, Output, State, html

alert = html.Div(
    [
        dbc.Button(
            "Toggle", id="alert-toggle-auto", className="me-1", n_clicks=0
        ),
        html.Hr(),
        dbc.Alert(
            "Hello! I am an auto-dismissing alert!",
            id="alert-auto",
            is_open=True,
            duration=4000,
        ),
    ]
)


@app.callback(
    Output("alert-auto", "is_open"),
    [Input("alert-toggle-auto", "n_clicks")],
    [State("alert-auto", "is_open")],
)
def toggle_alert(n, is_open):
    if n:
        return not is_open
    return is_open
